File: Anul 2/Semestrul 2/MPP/labs/Tema 5-proiect Server Csharp-Client Java/proiectSCcsharp/angajatServices/IService.cs
using proiectSCcsharp.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace proiectSCcsharp.services
{
    public interface IService
    {
        Angajat login(Angajat employee, IObserver employeeObserver);
        void logout(Angajat employee);
        Spectacol getSpectacol(long id);
        List<Spectacol> getAllSpectacole();
        void addBilet(Bilet bilet);

    }
}

File: Anul 2/Semestrul 2/MPP/labs/Tema 7-servicii REST/Java/Rest/proiect/Services/src/main/java/proiect/service/IService.java
package proiect.service;

import proiect.domain.Angajat;
import proiect.domain.Bilet;
import proiect.domain.Spectacol;
import proiect.domain.validators.ValidationException;

public interface IService {
    Spectacol getSpectacol(int id) throws MyException;
    Iterable<Spectacol> getAllSpectacole() throws MyException;
    void addBilet(Bilet bilet) throws MyException, ValidationException;
    Angajat login(Angajat employee, IObserver client) throws Exception;
    void logout(Angajat employee, IObserver client) throws Exception;
}
